ods_ci/libs/DataSciencePipelinesAPI.py

```diff
@@ -0,0 +1,235 @@
+import subprocess
+from json import JSONDecodeError
+import requests
+import os
+import json
+import time
+from robotlibcore import keyword
+import base64
+
+
+class DataSciencePipelinesAPI:
+
+    def __init__(self):
+        self.auth_url = self.retrieve_auth_url()
+        self.route = ''
+        self.sa_token = None
+
+    @keyword
+    def create_operator(self, file):
+        print(os.path.abspath(__file__))
+        current_path = __file__.index('/libs/')
+        current_path = __file__[:current_path]
+        response, error = self.run_oc(f"oc apply -f {current_path}/{file}")
+        print(f'create_operator({error}): {response}')
+
+    @keyword
+    def get_redhat_openshift_pipelines(self):
+        """
+        when creating at the first time, it can take like 1 minute to have the pods ready
+        """
+        deployment_count = 0
+        while deployment_count != 1:
+            deployments = []
+            response, _ = self.run_oc(f"oc get deployment -n openshift-operators openshift-pipelines-operator -o json")
+            try:
+                response = json.loads(response)
+                if response['metadata']['name'] == 'openshift-pipelines-operator' and 'readyReplicas' in response['status'] and response['status']['readyReplicas'] == 1:
+                    deployments.append(response)
+            except JSONDecodeError:
+                pass
+            deployment_count = len(deployments)
+            time.sleep(1)
```

Comment:
maybe you could use [oc wait](https://docs.openshift.com/container-platform/4.8/cli_reference/openshift_cli/developer-cli-commands.html#oc-wait) to spare a bit of lines

Reply:
I believe `oc wait` doesn't support nested AND or OR
also, https://github.com/kubernetes/kubernetes/issues/83242

Reply:
Probably you can avoid checking "response['metadata']['name'] == 'openshift-pipelines-operator'" because you're already getting it. So you may not need nested AND or OR.

Anyway, just a suggestion, I think it's okay what you implemented